task/upload.go

```diff
@@ -677,7 +675,11 @@ func uploadTaskOutputLocations(tctx *TaskContext) ([]OutputName, []clients.Outpu
 	} else {
 		mp4 = OUTPUT_ONLY_SHORT
 	}
-	outputNames, outputLocations, err := outputLocations(outURL, OUTPUT_ENABLED, playbackId, mp4, playbackId, "", "", !isEncryptionEnabled(*tctx.Task.Params.Upload))
+	thumbsEnabled := ""
+	if tctx.Task.Params.Upload.Thumbnails {
+		thumbsEnabled = OUTPUT_ENABLED
+	}
+	outputNames, outputLocations, err := outputLocations(outURL, out(OUTPUT_ENABLED, playbackId), out(mp4, playbackId), output{}, out(thumbsEnabled, playbackId), !isEncryptionEnabled(*tctx.Task.Params.Upload))
```

Comment:
nit: Mind break this down 1 arg per line so its more readable? Like on `TaskTranscodeFile`